Feature/tcin 60

GTerigiHastega opened the pull request:
Caching of POST, PUT and PATCH requests implemented. Need to refactor common code and further expand each and every method.

Commit:
Added nocache call

Zenomante committed:
implemented API call for cache cleaning

Zenomante committed:
Implemented force clear cache

Zenomante committed:
Changed force clear to folder clear

Zenomante committed:
Added manager

GTerigiHastega committed:
ESlint Fix on localCache
Post call working. Need to find a proper way to cache multiple <req.body, fetch.data> maps inside a single post.XX.json file.

GTerigiHastega committed:
ESlint Fix on localCache
Post call working. Need to find a proper way to cache multiple <req.body, fetch.data> maps inside a single post.XX.json file.

Zenomante committed:
Added swagger

GTerigiHastega committed:
Post, Put and Patch cache done.
Need to refractor and define common part of the code and extract it consequentially

GTerigiHastega committed:
Post, Put and Patch cache done.
Need to refractor and define common part of the code and extract it consequentially

pablomichelettii committed:
Merge pull request #3 from hastega/feature/TCIN1-67

Feature/tcin1 67

GTerigiHastega committed:
Merge remote-tracking branch 'origin/feature/TCIN-60' into feature/TCIN-60

GTerigiHastega committed:
Auto stash before merge of "feature/TCIN-60" and "origin/feature/TCIN-60"

GTerigiHastega committed:
New Manager, Refactored localCache.controller

New manager implemented for FS and process.env

Localcache.controller refactored.

pablomichelettii commented on src/app.ts:
```diff
@@ -21,17 +19,21 @@ import https from 'https';
-    const httpsAgent = new https.Agent({
-        rejectUnauthorized: false,
-    });
+// Manager
```
Don't use repetitve comments. If u use self explanatory naming convention, as u have done, is enough

pablomichelettii commented on src/app.ts:
```diff
@@ -21,17 +19,21 @@ import https from 'https';
+
+// To discuss
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const swaggerDocument = require('./swagger-output.json');
```
I suppose the problem is that u're trying to use a json importing it as a module. If not let's discuss what is the problem @Zenomante

pablomichelettii commented on src/app.ts:
```diff
@@ -21,17 +19,21 @@ import https from 'https';
-    console.log(process.env.NODE_ENV, `RejectUnauthorized is disabled.`);
+if (processEnvManager.isDevelopment()) {
+    axios.defaults.httpsAgent = new https.Agent({ rejectUnauthorized: false });
+    console.log('development', `RejectUnauthorized is disabled.`);
```
could we make this log dynamic?

pablomichelettii commented on src/controllers/localCache.controller.ts:
```diff
@@ -1,131 +1,121 @@
-            message: messageData,
-            data: responseData,
+    if (!noCache && existsSync(directoryPath) && existsSync(currentFile)) {
+        console.log(currentFile);
```
remove log

pablomichelettii commented on src/controllers/localCache.controller.ts:
```diff
@@ -1,131 +1,121 @@
+        successResponseBody.cache = false;
+        successResponseBody.filenameCached = currentFile;
     }
+    console.log(successResponseBody);
```
remove console log

pablomichelettii reviewed (changes requested):
that's nice! GG. For the next tasks remember to update the version of Loki (@Zenomante @GTerigiHastega )

GTerigiHastega committed:
New Logger for development purpose.

logger utility. 2 channels (console and file) for logging.
Added an utility that clears old logs files.

GTerigiHastega committed:
version 1.0.01

GTerigiHastega commented:
Logging utility added. Increased version to 1.0.01

GTerigiHastega committed:
Added error handling for axios response

GTerigiHastega committed:
catch errors on all axios calls

pablomichelettii commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "loki",
-  "version": "1.0.0",
+  "version": "1.0.01",
```
don`t use 0 before the number, 1.0.1 it's ok

pablomichelettii commented on package.json:
```diff
@@ -35,10 +37,13 @@
     "morgan": "^1.10.0",
     "nanoid": "^3.3.1",
     "node-typescript": "^0.1.3",
+    "npmlog": "^7.0.1",
```
you can remove it if u use wiston

pablomichelettii commented on src/app.ts:
```diff
@@ -21,17 +19,22 @@ import https from 'https';
+import processEnvManager from './manager/processEnv/processEnv.manager';
+import { logger } from './utils/logger';
+
+// To discuss
```
Choose a common tag for those comments: for example i'm pretty sure there are some vscode extensions that recognize tags like //TODO: or //@TODO: just chose one so we can start to use one of those tags

pablomichelettii commented on src/server.ts:
```diff
@@ -1,8 +1,11 @@
 
 app.listen(app.get('port'), () => {
-    console.log('Server on port', app.get('port'));
+    // console.log('Server on port', app.get('port'));
```
duplicate

pablomichelettii reviewed (commented):
GGWD

GTerigiHastega committed:
Pull request changes.

Removed console.log
Removed npmlog
Fixed version number

pablomichelettii reviewed (commented):
LGTM
